internal/manifests/collector/container.go

```diff
@@ -45,6 +45,10 @@ func Container(cfg config.Config, logger logr.Logger, agent v1alpha1.AmazonCloud
 
 	if addConfig {
 		volumeMounts = append(volumeMounts, getVolumeMounts(agent.Spec.NodeSelector["kubernetes.io/os"]))
+
+		if !agent.Spec.Prometheus.IsEmpty() {
```

Comment:
So we check before if prometheus is enabled before we create the volume but we don't check it before we volume mount it?

Reply:
We check in both cases.